lighthouse-core/lib/page-functions.js

```diff
@@ -400,6 +400,7 @@ function getNodeLabel(element) {
   }
 
   const tagName = element.tagName.toLowerCase();
+  const selector = getNodeSelector(element);
```

Comment:
two options here to avoid calling `getNodeSelector` twice per call to `getNodeDetails`:

1) have this function return null if no useful label, and use result of `getNodeSelector` as fallback in `getNodeDetails` (`getNodeLabel(...) || selector`)
2) have `getNodeSelector` accept a `default` parameter and return that as default

Reply:
Yeah that makes sense, thanks for the review! I will implement the changes you requested shortly.

Reply:
also you'll likely need to fix the tests (run `yarn jest page-functions`)

Reply:
Added the requested changes discussed in option 1, and yep two tests failed after running `yarn jest page-functions`

```
✕ Uses tag name for html tags (7 ms)
assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      "html"
    Received:
      null
✕ Uses tag name if there is no better label (2 ms)
...
```